### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by EmmanuelleB985**
@martinjrobins 
Hi Martin,
The tests are still failing remotely.
#include <nlohmann/json.hpp>  raises this error:
#include errors detected. Consider updating your compile_commands.json or includePath. Squiggles are disabled for this translation unit

I included the path of the json library in the configurations c_cpp_properties.json and settings.json 
This removes the error locally but when I push the tests fail.

### CODE REVIEW

**Review by martinjrobins — CHANGES_REQUESTED** [review]
sorry, I can't see the changes because it looks like you have accidentally committed a lot of files you didn't need (its showing 4128 files changed!). Can you clean this up  first?

**Comment by EmmanuelleB985** [comment]
> sorry, I can't see the changes because it looks like you have accidentally committed a lot of files you didn't need (its showing 4128 files changed!). Can you clean this up first?

We deleted dcmtk, it should fix the issue

File: src/parser.cpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#include <fstream>
+#include <nlohmann/json.hpp>
+#include <dcmtk/dcmpstat/dcmpstat.h>
+
+using json = nlohmann::json;
+
+using namespace std;
+
+
+class Instances {
```
Comment (martinjrobins) [inline src/parser.cpp:13]: the declaration of the class should go in a header file, e.g. `Parser.hpp`. And the name of the class should correspond to the name of the file, so in this case the source file name is `Parser.cpp` and the class name is `Parser`

File: src/parser.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    public:
+        void initial_settings();
```
Comment (martinjrobins) [inline src/parser.cpp:16]: can you add comments giving a description of what you think each of these functions should do? it doesn't matter if you haven't implemented all the functionality yet, just give the plan. Ideally in doxygen style (https://www.doxygen.nl/manual/docblocks.html), but this isn't essential at this stage

File: src/parser.cpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        void function_extractor(); 
+};
+
+void Instances::initial_settings(){
+
+    //open json config file 
+    //std::ifstream file1("/home/sabsr3/ECIQC/src/draft_user_config.json");
```
Comment (martinjrobins) [inline src/parser.cpp:23]: just remove commented out code, it is still in version control so its not lost

File: tests/MyTests.cpp [hunk 4]
```diff
@@ -41,6 +41,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "../src/Exception.hpp"
 #include "../src/MetadataEditor.cpp"
 #include "../src/MetadataEditor.hpp"
+#include "../src/parser.cpp"
```
Comment (martinjrobins) [inline tests/MyTests.cpp:44]: you should include the header file here (eg. `Parser.hpp`)

**Review by martinjrobins — CHANGES_REQUESTED** [review]
Thanks @EmmanuelleB985, this is a great start. I've made a few suggestions, and there are
few files that need removing (I can't see that they are used anywhere?)
1-003.dcm
1-004.dcm
1-005.dcm
1-006.dcm
dcmtk-3.6.1_20170228.tar.gz
src/parser
output/tags.json
output/values.json
tests/test

**Comment by EmmanuelleB985** [comment]
> Thanks @EmmanuelleB985, this is a great start. I've made a few suggestions, and there are
> few files that need removing (I can't see that they are used anywhere?)
> 1-003.dcm
> 1-004.dcm
> 1-005.dcm
> 1-006.dcm
> dcmtk-3.6.1_20170228.tar.gz
> src/parser
> output/tags.json
> output/values.json
> tests/test

Thank you for the suggestions! Emmanuel sent me these .dcm files this morning but I haven't used them yet and we can remove the other files as well

File: src/parser.cpp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    \It includes basic functionalities for the moment. 
+*/
+void Parser::initial_settings(){
+
+    
+    auto js = R"({"GCC_compiler_version":"9.3.0",
+    "warn_return_any":"True",
+    "warn_unused_configs":"True",
+    "ignore_missing_imports":"True",
+    "include_private_tags":"True"})"_json; /**< create a string containing configuration files and convert it to a json object */
```
Comment (PavanChaggar) [inline src/parser.cpp:24]: Would it be better to move the test json to MyTests.cpp and have it given as input to the parser?

Comment (EmmanuelleB985) [reply src/parser.cpp:24]: Yes we want to give the test json as input to the parser. The stringstream from c++ did not integrated well with the json library as opposed to the file stream so I used temporarily raw string literals.

**Comment by martinjrobins** [comment]
hi @EmmanuelleB985, the error that you are getting is the same as previous, you are not able to read in the json file as the test is run from a different directory. 

You should avoid reading in any existing files, instead write your parser class to that it takes in a stream which contains the configuration. I've made a few edits in a different branch `parser2`, which demonstrates this.

**Pull request merged**